client/components/search.js

```diff
@@ -0,0 +1,60 @@
+import React, {Component} from 'react'
+import {Input, Button} from 'semantic-ui-react'
+import store from '../store/index.js'
+import {connect} from 'react-redux'
+import {setResults} from '../store/search'
+
+class Search extends Component {
+  constructor() {
+    super()
+    this.state = {
+      searchString: ''
+    }
+  }
+  handleChange = evt => {
+    this.setState({
+      searchString: evt.target.value
+    })
+  }
+  handleSubmit = evt => {
+    evt.preventDefault()
+
+    const resultArray = this.createSearchResults(this.state.searchString)
+    this.props.setResults(resultArray)
+  }
+  createSearchResults = searchString => {
+    const state = store.getState()
+
+    const results = state.products.allProducts.filter(product => {
```

Comment:
Algorithm itself looks great. I think for now we can keep this logic in the component (I'm doing the same thing with the filter), but eventually I think it makes sense to migrate to the store